ocs_ci/ocs/resources/packagemanifest.py

```
@@ -307,3 +308,93 @@ def get_selector_for_ocs_operator():
+        f'.status.catalogSource == "{safe_catalog_source}")'
+    )
+
+    # Pipe oc command directly to yq to reduce memory usage
+    cmd = (
+        f"oc {kubeconfig_flag}get packagemanifest "
+        f"-n {constants.MARKETPLACE_NAMESPACE} -o yaml | "
+        f"yq eval '{yq_filter}' -"
+    )
+
+    try:
+        result = exec_cmd(cmd, shell=True)
```

Ruff (0.15.20)

[error] Function call with `shell=True` parameter identified, security issue

(S604)